com.microsoft.mrtk.uxcomponents/Tests/Runtime/NonNativeKeyboardTests.cs

```diff
@@ -0,0 +1,350 @@
+		{
+			testKeyboard.CloseOnInactivity = true;
+			testKeyboard.CloseOnInactivityTime = .01f;
+
+			NonNativeValueKey akey = SetUpValueKey("a");
+			Button Ainteractable = akey.gameObject.GetComponent<Button>();
+			Ainteractable.onClick.Invoke();
+
+
+			yield return new WaitForSeconds(0.015f);
+
+			Assert.AreEqual(inputfield.text, "", "The input field is cleared.");
+			Assert.IsTrue(!testKeyboard.gameObject.activeInHierarchy, "The keyboard is inactive.");
+			yield return null;
+		}
+
+		[UnityTest]
+		public IEnumerator TestNonnativeValueKeyPressKeyWithHand()
+		{
+			NonNativeValueKey nkey = SetUpValueKey("n");
+			nkey.transform.position = new Vector3(0f, 1.6f, .69f);
+			yield return RuntimeTestUtilities.WaitForUpdates();
+
+			TestHand hand = new TestHand(Handedness.Right);
+			Vector3 initialHandPosition = InputTestUtilities.InFrontOfUser(new Vector3(0.05f, -0.05f, 0.3f)); // orient hand so far interaction ray will hit button
+			yield return hand.Show(initialHandPosition);
+			yield return RuntimeTestUtilities.WaitForUpdates();
+			yield return hand.SetHandshape(HandshapeId.Pinch);
+			yield return RuntimeTestUtilities.WaitForUpdates();
+			yield return hand.SetHandshape(HandshapeId.Open);
+			yield return RuntimeTestUtilities.WaitForUpdates();
+
+			yield return RuntimeTestUtilities.WaitForUpdates();
+
+			Assert.AreEqual(inputfield.text, "n", "Pressing key changes InputField text.");
+			yield return null;
+		}
+
+		private NonNativeValueKey SetUpValueKey(string value)
+		{
+			NonNativeValueKey[] keys = testKeyboard.gameObject.GetComponentsInChildren<NonNativeValueKey>(true);
+			foreach (NonNativeValueKey key in keys)
+			{
+				if (key.CurrentValue?.Equals(value) == true) return key;
+			}
+			return null;
+		}
+
+		private NonNativeFunctionKey SetUpFunctionKey(NonNativeFunctionKey.Function function)
+		{
+			NonNativeFunctionKey[] keys = testKeyboard.gameObject.GetComponentsInChildren<NonNativeFunctionKey>(true);
+			foreach (NonNativeFunctionKey key in keys)
+			{
+				if (key.KeyFunction == function) return key;
+			}
+			return null;
+		}
+
+		private void PressFunctionKey(NonNativeFunctionKey function)
+		{
+			NonNativeValueKey akey = SetUpValueKey("a");
+			NonNativeValueKey bkey = SetUpValueKey("b");
+
+			Button Ainteractable = akey.gameObject.GetComponent<Button>();
+			Button Binteractable = bkey.gameObject.GetComponent<Button>();
+			Ainteractable.onClick.Invoke();
+			testKeyboard.ProcessFunctionKeyPress(function);
+			Binteractable.onClick.Invoke();
+		}
+
+		private GameObject InstantiatePrefab(string prefabPath)
+		{
+			Object pressableButtonPrefab = AssetDatabase.LoadAssetAtPath(prefabPath, typeof(Object));
+			GameObject testGO = Object.Instantiate(pressableButtonPrefab) as GameObject;
+
+			return testGO;
+		}
+
+		/// <summary>
+		/// Move the hand forward to press button, then off to the right to release
+		/// </summary>
+		private IEnumerator PressAndReleaseButtonWithHand(Vector3 buttonPosition)
```

Comment:
Please add something to verify that the keyboard's text changed with hand input

Sorry if I missed it...but I don't see anything calling PressAndReleaseButtonWithHand

Reply:
Oh I see `TestNonnativeValueKeyPressKeyWithHand` now....please you delete `PressAndReleaseButtonWithHand` and it's other functions....I assume they aren't needed anymore

Reply:
just deleted, thanks